apps/hugging-face/src/locations/Page.tsx

```diff
@@ -1,35 +1,33 @@
+import { useState, useEffect } from 'react';
 import { PageAppSDK } from '@contentful/app-sdk';
-import { Grid, Form, FormControl, Textarea, Button, Flex, Heading, Paragraph, Stack, Note } from '@contentful/f36-components';
 import { useSDK } from '@contentful/react-apps-toolkit';
-import { useState, useEffect } from 'react';
-import css from '@emotion/css';
 import { generateImage } from '../services/huggingfaceImage';
 import { refinePrompt } from '../services/huggingfaceText';
-
-interface AppInstallationParameters {
-  huggingfaceApiKey?: string;
-  textModelId?: string;
-  imageModelId?: string;
-}
+import { AppInstallationParameters } from '../utils/types';
+import { GenerateImageModal } from '../components/GenerateImageModal';
+import { RefinePromptModal } from '../components/RefinePromptModal';
+import { InitialPrompt } from '../components/InitialPrompt';
 
 const Page = () => {
   const sdk = useSDK<PageAppSDK>();
   const [initialPrompt, setInitialPrompt] = useState('');
   const [isRefining, setIsRefining] = useState(false);
+  const [refinedPrompt, setRefinedPrompt] = useState('');
   const [isGenerating, setIsGenerating] = useState(false);
   const [generatedImage, setGeneratedImage] = useState<string | null>(null);
   const [error, setError] = useState<string | null>(null);
-  const [refinedPrompt, setRefinedPrompt] = useState('');
-  const [isUploading, setIsUploading] = useState(false);
+  // const [isUploading, setIsUploading] = useState(false);
   const [timer, setTimer] = useState(0);
   const [isTimerActive, setIsTimerActive] = useState(false);
+  const [showRefinePromptModal, setShowRefinePromptModal] = useState(false);
+  const [showGeneratingImageModal, setShowGeneratingImageModal] = useState(false);
```

Comment:
I thought of a small idea here that might make this more scalable in case we have more modals to add.

Assuming we only want to show one modal at a time, instead of having individual boolean state for each modal this could use one state variable that has values like `'refine-prompt' | 'generating-image' | null` where null means no modals are showing, and a string means that modal is active.